strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-npm-layout/strongbox-npm-metadata-jackson/src/main/java/org/carlspring/strongbox/npm/metadata/jackson/BinDeserializer.java

```diff
@@ -0,0 +1,42 @@
+package org.carlspring.strongbox.npm.metadata.jackson;
+
+import java.io.IOException;
+
+import org.carlspring.strongbox.npm.metadata.Bin;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.ObjectCodec;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+
+public class BinDeserializer extends JsonDeserializer<Bin>
+{
+
+    @Override
+    public Bin deserialize(JsonParser jp,
+                           DeserializationContext c)
+        throws IOException,
+        JsonProcessingException
+    {
+
+        ObjectCodec codec = jp.getCodec();
+        JsonNode node = codec.readTree(jp);
+
+        String value = node.asText();
+
+        if (value.startsWith("{"))
+        {
+            return codec.treeToValue(node, Bin.class);
+        }
+
+        Bin bin = new Bin();
+        //TODO: we should replace this {packageName} placeholer with package name.
+        bin.setAdditionalProperty("{packageName}", value);
```

Comment:
@sbespalov : Do we need a follow-up task for this? If yes, could you please clarify what needs to be done? If not and this is no longer relevant, should we remove the `// TODO`?

Reply:
I don't remember what this was about :)